### PR TITLE
feat(team-workflow): Add subscription for team in `GroupSubscription`

### PR DESCRIPTION
**Pull request by isabellaenriquez**
Updates `subscribe_actor` to no longer read all the team's users and instead subscribe the Team itself if the team-workflow-notifications feature flag is enabled. Also updates `subscribe` so that it can accept a Team as a subscriber, not just Users or RpcUsers.

Part 1 of #55556 

~~❗ BLOCKED BY #55879~~

### CODE REVIEW

**Comment by isabellaenriquez** [comment]
there's gonna be one failing test bc we need the GroupSubscription migration 😬

File: tests/sentry/models/test_groupsubscription.py [hunk 1]
```diff
@@ -22,12 +22,12 @@ def test_simple(self):
         group = self.create_group()
         user = self.create_user()
```
Comment (roggenkemper) [inline tests/sentry/models/test_groupsubscription.py:23]: would be good to add a simple test case here where the subscriber is the team

File: tests/sentry/models/test_groupsubscription.py [hunk 2]
```diff
@@ -88,6 +97,23 @@ def test_actor_team(self):
         # should not error
         GroupSubscription.objects.subscribe_actor(group=group, actor=team)
 
+    @with_feature("organizations:team-workflow-notifications")
+    def test_subscribe_team(self):
```
Comment (roggenkemper) [inline tests/sentry/models/test_groupsubscription.py:101]: nice! good test to add in!

File: src/sentry/models/groupsubscription.py [hunk 3]
```diff
@@ -34,22 +34,33 @@ class GroupSubscriptionManager(BaseManager):
...
-                )
+                if isinstance(subscriber, User) or isinstance(subscriber, RpcUser):
+                    self.create(
+                        user_id=subscriber.id,
+                        group=group,
+                        project=group.project,
+                        is_active=True,
+                        reason=reason,
+                    )
+                elif isinstance(subscriber, Team):
```
Comment (scefali) [inline src/sentry/models/groupsubscription.py:56]: @isabellaenriquez what about `RpcTeam`?

Comment (isabellaenriquez) [reply src/sentry/models/groupsubscription.py:56]: good catch!

Comment (isabellaenriquez) [reply src/sentry/models/groupsubscription.py]: ok update @scefali, i just realized that there doesn't seem to be a way to unpack team members from an RpcTeam instance, which is an issue when the feature flag isn't enabled (RpcTeam was never included in the first place, so this wasn't dealt with before).

**Review by scefali — APPROVED** [review]

File: src/sentry/models/groupsubscription.py (hunk 3 above)
Comment (isabellaenriquez) [reply src/sentry/models/groupsubscription.py]: There doesn't appear to be any instances where `subscribe_actor` would be called with an RpcTeam either.

**Pull request merged**